include/circt/Dialect/ESI/ESIPasses.td

```diff
@@ -14,6 +14,12 @@ def LowerESIToPhysical: Pass<"lower-esi-to-physical", "mlir::ModuleOp"> {
   let constructor = "circt::esi::createESIPhysicalLoweringPass()";
 }
 
+def LowerESIPorts: Pass<"lower-esi-ports", "mlir::ModuleOp"> {
+  let summary = "Lower ESI input and/or output ports.";
+  let constructor = "circt::esi::createESIPortLoweringPass()";
+  let dependentDialects = ["circt::rtl::RTLDialect"];
```

Comment:
does this need to depend on RTL? i think dialects that can be part of the input _don't_ need to be declared here (they should be loaded lazily as the IR is parsed). does this pass encounter RTL ops/types/attributes before the dialect had a chance to get loaded?

also, should this depend on the SV dialect? i think it does, since it may generate SV interface ops.

Reply:
Good catch. I was thinking that since it works on externmodules, it depended on the rtl dialect, but you're right.